ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -314,6 +326,26 @@ public void SetBackgroundColor(BorderedCanvas view, uint color)
             }
         }
 
+        /// <summary>
+        /// Gets background brush of Canvas itself or of Border child as
+        /// appropriate. This is the only supported method to get background
+        /// of BorderedCanvas. DO NOT get view.Background or Border.Background directly.
```

Comment:
> DO NOT get view.Background or Border.Background [](start = 30, length = 48)

Would it be better to put all this logic into `BorderedCanvas`?

Reply:
It would be in my opinion. But this would require moving all the "optional border" logic from manager to view as well. I will start working on such a change. Please tell me if you think it's too much of a change though.

---
In reply to: [177187283](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177187283) [](ancestors = 177187283)

Reply:
this would also require to expose the functionality as public (or internal) instead of private/protected now, which extends public API surface of BorderdCanvas.

---
In reply to: [177272760](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177272760) [](ancestors = 177272760,177187283)

Reply:
@zholobov - would you still have to extend `ReactViewManager` in addition to the `BorderedCanvas` for linear gradient, or would you just create a new view manager implementation?

Reply:
@rozele I guess not since border radius logic and children management would also go to BorderedCanvas since it is related on details of implementation of optional border. So ReactViewManager would be sort of empty just forwarding setters to BorderedCanvas.